### PR TITLE
Add Besu Execution Client

### PR DESCRIPTION
**Pull request by salaheldinsoliman**
## Changes:
- Added Besu required template
## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests?**

- [ ] Yes
- [x] No

Need to run smoke tests

### CODE REVIEW

File: templates/services/merge/execution/besu.tmpl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - --rpc-http-enabled=true
+    - --rpc-http-host=*
+    - --rpc-http-port={{.ElApiPort}}
+    - --rpc-http-api=ETH,NET,CLIQUE,DEBUG,MINER,NET,PERM,ADMIN,EEA,TXPOOL,PRIV,WEB3
+    - --rpc-http-cors-origins=*
+
+
+    - --host-whitelist=*
+
+
```
Comment (AntiD2ta) [inline templates/services/merge/execution/besu.tmpl:46]:
```suggestion
    - --metrics-enabled=true
    - --metrics-host=0.0.0.0
    - --metrics-port={{.ElMetricsPort}}
    - --engine-rpc-enabled=true
    - --engine-jwt-secret=/var/lib/besu/jwtsecret
    - --engine-rpc-port={{.ElAuthPort}}
    - --engine-host-allowlist=*
    - --rpc-http-enabled=true
    - --rpc-http-host=*
    - --rpc-http-port={{.ElApiPort}}
    - --rpc-http-api=ETH,NET,CLIQUE,DEBUG,MINER,NET,PERM,ADMIN,EEA,TXPOOL,PRIV,WEB3
    - --rpc-http-cors-origins=*
    - --host-whitelist=*
```

File: templates/envs/goerli/execution/besu.tmpl [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{{/* besu.tmpl */}}
+{{ define "execution" }}
+# --- Execution Layer - Execution Node - configuration ---
+EC_IMAGE_VERSION={{if .ElImage}}{{.ElImage}}{{else}}hyperledger/besu{{end}}
```
Comment (AntiD2ta) [inline templates/envs/goerli/execution/besu.tmpl:4]:
```suggestion
EC_IMAGE_VERSION={{if .ElImage}}{{.ElImage}}{{else}}hyperledger/besu:22.7.4{{end}}
```

File: templates/envs/mainnet/execution/besu.tmpl [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{{/* besu.tmpl */}}
+{{ define "execution" }}
+# --- Execution Layer - Execution Node - configuration ---
+EC_IMAGE_VERSION={{if .ElImage}}{{.ElImage}}{{else}}hyperledger/besu{{end}}
```
Comment (AntiD2ta) [inline templates/envs/mainnet/execution/besu.tmpl:4]:
```suggestion
EC_IMAGE_VERSION={{if .ElImage}}{{.ElImage}}{{else}}hyperledger/besu:22.7.4{{end}}
```

File: templates/envs/sepolia/execution/besu.tmpl [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{{/* besu.tmpl */}}
+{{ define "execution" }}
+# --- Execution Layer - Execution Node - configuration ---
+EC_IMAGE_VERSION={{if .ElImage}}{{.ElImage}}{{else}}hyperledger/besu{{end}}
```
Comment (AntiD2ta) [inline templates/envs/sepolia/execution/besu.tmpl:4]:
```suggestion
EC_IMAGE_VERSION={{if .ElImage}}{{.ElImage}}{{else}}hyperledger/besu:22.7.4{{end}}
```

File: templates/services/merge/execution/besu.tmpl [hunk 5]
```diff
@@ -0,0 +1,41 @@
+{{/* besu.tmpl */}}
+{{ define "execution" }}
+  execution:
+    image: ${EC_IMAGE_VERSION}
+    container_name: execution_client
```
Comment (AntiD2ta) [inline templates/services/merge/execution/besu.tmpl:5]: Please put the properties in the same order as in the other service templates.

File: templates/services/merge/execution/besu.tmpl [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    - --engine-rpc-port={{.ElAuthPort}}
+    - --engine-host-allowlist=*
+    - --rpc-http-enabled=true
+    - --rpc-http-host=0.0.0.0
+    - --rpc-http-port={{.ElApiPort}}
+    - --rpc-http-api=ETH,NET,CLIQUE,DEBUG,MINER,NET,PERM,ADMIN,EEA,TXPOOL,PRIV,WEB3
+    - --rpc-http-cors-origins=*
+    - --host-whitelist=*
+    restart: unless-stopped
+    mem_swappiness: 0
```
Comment (AntiD2ta) [inline templates/services/merge/execution/besu.tmpl:40]: I believe putting `mem_swappiness` here and not within a `resources` property is deprecated in Compose v3.0. Is there a reason why we should configure this?

File: templates/services/merge/execution/besu.tmpl [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    - --metrics-host=0.0.0.0
+    - --metrics-port={{.ElMetricsPort}}
+    - --engine-rpc-enabled=true
+    - --engine-jwt-secret=/var/lib/besu/jwtsecret
+    - --engine-rpc-port={{.ElAuthPort}}
+    - --engine-host-allowlist=*
+    - --rpc-http-enabled=true
+    - --rpc-http-host=0.0.0.0
+    - --rpc-http-port={{.ElApiPort}}
+    - --rpc-http-api=ETH,NET,CLIQUE,DEBUG,MINER,NET,PERM,ADMIN,EEA,TXPOOL,PRIV,WEB3
```
Comment (AntiD2ta) [inline templates/services/merge/execution/besu.tmpl:36]:
```suggestion
    - --rpc-http-api=${EC_ENABLED_MODULES}
```

**Review by AntiD2ta — CHANGES_REQUESTED** [review]
Please update the README with this new client, specially the supported network tables. Also update /scripts/check-image-updates.sh with Besu, so we can get the latest release from Besu as soon as possible. Very good feature, some changes requested though.

File: templates/services/merge/execution/besu.tmpl (hunk 6 above)
Comment (stdevMac) [reply templates/services/merge/execution/besu.tmpl:47]: No reason, deleting it now

File: templates/services/merge/execution/besu.tmpl (hunk 5 above)
Comment (stdevMac) [reply templates/services/merge/execution/besu.tmpl]: done

**Review by AntiD2ta — APPROVED** [review]

**Pull request merged**